cmd/bootstrapper/generator.go

```diff
@@ -56,23 +57,42 @@ func WithFilesystem(fs afero.Fs) Opt {
 	}
 }
 
-func WithHttpClient(c *http.Client) Opt {
+func WithHttpClient(c *retryablehttp.Client) Opt {
 	return func(g *Generator) {
 		g.client = c
 	}
 }
 
+func checkRetry(ctx context.Context, resp *http.Response, err error) (bool, error) {
+	if resp != nil && resp.StatusCode != http.StatusOK {
+		return true, nil
+	}
+	return retryablehttp.DefaultRetryPolicy(ctx, resp, err)
+}
```

Comment:
I think just the default retry policy is sufficient. It will retry on errors, http status 429 and any http status 5xx other 4xx or 3xx errors I would not expect to resolve after retrying.